### PR TITLE
pkg/ipmi, boot/systemboot: Send boot drive info to BMC

### PR DESCRIPTION
**Pull request by JingleHsuWiwynn**
1. Add pkg SMBIOS type 9 basic frame, including table type, data structure, and parsing utility.
2. With SMBIOS type 9 data, implement IPMI OEM command for sending boot drive's vendor and device id to BMC.

### CODE REVIEW

**Comment by johnnylinwiwynn** [comment]
@jonzhang-fb

File: cmds/boot/systemboot/main.go [hunk 1]
```diff
@@ -178,6 +178,17 @@ func runIPMICommands() {
 				log.Printf("Failed to get the information of Processors: %v.", err)
 			}
 
+			BootDriveInfo, err := ocp.GetOemIpmiBootDriveInfo(si)
+			if err == nil && BootDriveInfo != nil {
```
Comment (rminnich) [inline cmds/boot/systemboot/main.go:182]: if err != nil {
}
if BootDriveInfo != nil {
}
As it is, the code does not distinguish the error case from the no info case, and I think that's an important difference.

File: pkg/ipmi/ocp/ocp.go [hunk 2]
```diff
@@ -286,6 +318,52 @@ func GetOemIpmiDimmInfo(si *smbios.Info) ([]DimmInfo, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	t9, err := si.GetSystemSlots()
+	if err != nil {
+		return nil, err
+	}
+
+	const systemPath = "/sys/bus/pci/devices/"
```
Comment (rminnich) [inline pkg/ipmi/ocp/ocp.go:336]: This file probably needs to be named ocp_linux.go, since there are linux depdendencies like this in it.

**Review by rminnich — COMMENTED** [review]
I found a small issue with the if, but a larger issue: the ocp.go should be named ocp_linux.go, as it is linux specific.

**Comment by johnnylinwiwynn** [comment]
Thanks for the review, we will update the PR at a later time.

File: pkg/ipmi/ocp/ocp.go (hunk 2 above)
Comment (JingleHsuWiwynn) [reply pkg/ipmi/ocp/ocp.go:336]: Done

File: cmds/boot/systemboot/main.go (hunk 1 above)
Comment (JingleHsuWiwynn) [reply cmds/boot/systemboot/main.go]: Done

**Comment by johnnylinwiwynn** [comment]
> in SMBIOS version 3.5 spec, type 9 is now able to define whether a slot supports CXL 1.0 or 2.0 or both. See table 51/52. Do we detect the CXL slot and set it accordingly?

This PR handles type 9 tables that is a 'Boot_Drive' and send its information to BMC. Setting CXL related info to SMBIOS type 9 should be done by the bootloader before u-root such as coreboot.

**Review by MDr164 — APPROVED** [review]
lgtm

**Review by rminnich — APPROVED** [review]

**Pull request merged**